lib/mongo/write_concern/base.rb

```diff
@@ -43,8 +43,24 @@ class Base
       #
       # @since 2.0.0
       def initialize(options)
-        opts = Options::Mapper.transform_keys_to_symbols(options)
-        @options = Options::Mapper.transform_values_to_strings(opts).freeze
+        options = Options::Mapper.transform_keys_to_symbols(options)
+        options = Options::Mapper.transform_values_to_strings(options).freeze
+
+        if options[:w]
+          if options[:w] == 0 && options[:j]
+            raise Error::InvalidWriteConcern, "Invalid write concern options: :j cannot be true when :w is 0: #{options.inspect}"
+          elsif options[:w] == 0 && options[:fsync]
+            raise Error::InvalidWriteConcern, "Invalid write concern options: :fsync cannot be true when :w is 0: #{options.inspect}"
+          elsif options[:w].is_a?(Integer) && options[:w] < 0
+            raise Error::InvalidWriteConcern, "Invalid write concern options: :w cannot be negative (#{options[:w]}): #{options.inspect}"
+          end
```

Comment:
Do we also want to validate the types of the fields that are present (e.g. that `w` is either an integer, string, or symbol, j is a boolean, etc.)?

Reply:
I have not run into this issue to date.

Reply:
The idea would be for this to benefit users, who wouldn't otherwise have a silent failure. If you don't think that's helpful, we can leave it as-is